enhancements/support-less-on-more.md

```diff
@@ -0,0 +1,146 @@
+---
+status: propose
+title: Update OpenShift Pipelines support policy to support less versions on more OpenShift versions
+creation-date: '2023-07-31'
+last-updated: '2023-07-31'
+authors: ['vdemeester']
+---
+
+# Update OpenShift Pipelines support policy to support less versions on more OpenShift versions
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+- [Proposal](#proposal)
+  - [Support only the last 2 releases](#support-only-the-last-2-releases)
+  - [Support down to <code>MIN_KUBERNETES_VERSION</code>](#support-down-to-)
+- [Testing](#testing)
+- [Dependencies](#dependencies)
+- [Examples](#examples)
+- [When ?](#when-)
+<!-- /toc -->
+
+---
+type: project
+deadline: 2023-09-02
+tags:
+  - project
+---
+## Summary
+
+The plan is to reduce the number of OpenShift Pipelines version we
+support at a given time while augmenting the number of OpenShift
+version we support them in.
+
+
+Supporting Materials:
+ * [Enhance Support Policy of OpenShift Pipelines](./enhance-support-policy.md)
+ * [OCP 4.y lifecycles](https://docs.google.com/document/d/1exNNJPdwCH3F5eEKaLVBcrj_qAHfJ3PlfnquP0-6TxM/edit#heading=h.539ohzh6t1k6)
+ * [Proposal for Pipelines &
+   GitOps](https://docs.google.com/spreadsheets/d/1Tn6BHNBJnEaTBRm22U0KrNCR_Zwu_qBn-IR-xFEa4Os/edit#gid=1430910332)
+
+## Motivation
+
+OpenShift Pipelines versions are currently only shipped and supported
+for a small set of OpenShift version (n-1, n, n+1).  OpenShift support
+can be very long (especially for EUS version), which makes us having
+to support a lot of version at the same time. Adding to this support
+exception that we may give to customers, it can become a burden for
+maintenance, and a lot of bug-fix version to release.
+
+For example, today (2023, 31st of July), we are *technically*
+supporting the following:
+- 1.8.x on 4.10 (EUS)
+- 1.9.x on >= 4.10
+- 1.10.x
+- 1.11.x
+
+In addition, it has no real technological reason for not supporting
+multiple version. Tekton supports a relatively wide range of Kubernetes
+version, and so can OpenShift Pipelines.
+
+Enabling customers to get the latest of feature from Tekton in their
+supported cluster is a win-win situation : customers can use the
+latest, and we can ship the latest (fixes, …) to our customers.
+
+## Proposal
+
+### Support only the last 2 releases
+
+The proposal is to only support the last 2 releases. As of today, this
+would mean supporting 1.11 and 1.10.
+
+In addition, as soon as release N is released, release N-1 becomes in
+maintenance mode, aka gets only critical bugfixes and security
+fixes. Quick note on this, it would also get any bugfix release if
+upstream components tracked in that release do a bugfix release as
+well.
+
+The idea of the proposal is to make it more appealing to get the
+latest release. The only challenge can be the
+`MIN_KUBERNETES_VERSION`, as discussed below.
+
+### Support down to `MIN_KUBERNETES_VERSION`
```

Comment:
when a new OCP is released and minimum k8s version is satisfied for N-1 version, then do we release N-1 on the new OCP version as well?
for example: 1.11 and 1.12 are supported on 4.12, 4.13, 4.14; now when 4.15 is released, will 1.11 support 4.15 as well?

Reply:
Yes *if* there is not already 2 releases on that versions. In 4.15 case, we should already have 1.12 and 1.13, so we don't really need to add 4.15 to 1.11

Reply:
I think that will work *once* in the given use case.
Say time goes by and we have Pipelines 1.14 GA.
We would not want to de-support Pipelines 1.12 on OCP 4.15 at that time.
(Reason:  Say a customer is on OCP 4.12.  They have available Pipelines 1.11, 1.12.  They choose to upgrade OCP.
- The newest Pipelines they can be on is 1.12, they cannot upgrade to OCP 4.15, which will not support 1.12.  They are 'stuck')

Reply:
@RickJWagner if 1.12 is not "supported" on 4.15 it would *still* not block the upgrade as, technically it still runs, *and* in the "upgrade path", 1.14 comes from 1.13 which comes from 1.12, so the "chain" shouldn't be broken imo.

Reply:
decision: while shipping a new minor release, we take the minimum k8s version in account to add the supported OCP versions for that release - https://access.redhat.com/solutions/4870701